controllers/consoleplugin.go

```diff
@@ -0,0 +1,405 @@
+package controllers
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"context"
+
+	consolepluginv1 "github.com/openshift/api/console/v1alpha1"
+	pipelinesv1alpha1 "github.com/redhat-developer/gitops-operator/api/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	resourcev1 "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	gitopsPluginName             = "gitops-plugin"
+	displayName                  = "GitOps Plugin"
+	gitopsPluginSvcName          = gitopsPluginName + "-service"
+	proxyAlias                   = "gitops"
+	pluginImageEnv               = "GITOPS_CONSOLE_PLUGIN_IMAGE"
+	servicePort                  = 9001
+	pluginServingCertName        = "console-serving-cert"
+	kubeAppLabelApp              = "app"
+	kubeAppLabelComponent        = "app.kubernetes.io/component"
+	kubeAppLabelInstance         = "app.kubernetes.io/instance"
+	kubeAppLabelPartOf           = "app.kubernetes.io/part-of"
+	kubeAppLabelRuntimeNamespace = "app.kubernetes.io/runtime-namespace"
+	httpdConfigMapName           = "httpd-cfg"
+)
+
+func pluginDeployment() *appsv1.Deployment {
+	consolePluginImage := os.Getenv(pluginImageEnv)
+	if consolePluginImage == "" {
+		consolePluginImage = "quay.io/redhat-developer/gitops-console-plugin"
+	}
+	podSpec := corev1.PodSpec{
```

Comment:
Can you please test this pod specification/deployment on a 4.12 cluster which is based on k8s v1.25 to verify if we need to include podSecurity ?

https://kubernetes.io/docs/tasks/configure-pod-container/security-context/